### PR TITLE
[CORE-852] Re-apply dYdX changes from 0.50.1 on top of 0.50.3

### PR DESCRIPTION
**Pull request by lcwik**
## Description

Re-apply dYdX changes from 0.50.1 on top of 0.50.3

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] added `!` to the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] updated the relevant documentation or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] run `make lint` and `make test`
* [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage
* [ ] manually tested (if applicable)

### CODE REVIEW

File: server/start.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ is performed. Note, when enabled, gRPC will also be automatically enabled.
 			}
 
 			err = wrapCPUProfile(serverCtx, func() error {
-				return start(serverCtx, clientCtx, appCreator, withCMT, opts)
+				return start(cmd.Context(), serverCtx, clientCtx, appCreator, withCMT, opts)
```
Comment (teddyding) [inline server/start.go:175]: For my understanding why was this needed (asking because I don't see this change in https://github.com/dydxprotocol/cosmos-sdk/pull/30/files)

**Review by teddyding — APPROVED** [review]

File: server/start.go (hunk 1 above)
Comment (lcwik) [reply server/start.go:175]: It is to allow the caller to shutdown cosmos using the `context` that is associated with the command.

It was added later in https://github.com/dydxprotocol/cosmos-sdk/commit/b48250d4e0efacbd40f3e0b980c1e1004d29e351
